Sources/ZetaPushNetwork/Client/ClientHelper.swift

```diff
@@ -324,31 +294,35 @@ extension ClientHelper: CometdClientDelegate {
     
     delegate?.onSuccessfulHandshake(self)
   }
-  
-  public func handshakeDidFailed(from client: CometdClientContract) {
+
+  public func handshakeDidFailed(error: CometDClientError, from client: CometdClientContract) {
     log.zp.error("ClientHelper Handshake Failed")
-    delegate?.onFailedHandshake(self)
+    delegate?.onFailedHandshake(self, error: .handshakeFailed(error: error))
   }
   
-  public func didDisconnected(error: Error?, from client: CometdClientContract) {
+  public func didDisconnected(error: CometDClientError?, from client: CometdClientContract) {
     log.zp.debug("ClientHelper Disconnected from Cometd server")
     connected = false
-    delegate?.onConnectionClosed(self)
+    if let error = error {
+      delegate?.onConnectionClosed(self, error: .connectionClosed(error: error))
+    } else {
+      delegate?.onConnectionClosed(self, error: nil)
```

Comment:
Here you set nil to error, do you think about putting .unknown connection close error instead ?

Reply:
A connection can be closed without error